http/src/v0/dht.rs

```diff
@@ -144,17 +145,16 @@ async fn provide_query<T: IpfsTypes>(
         verbose: _,
         timeout,
     } = query;
-    let key = arg.into_bytes();
-    ipfs.provide(key)
+    let cid = arg.parse::<Cid>().map_err(StringError::from)?;
```

Comment:
is there an expected error message in the tests which forbids the use of `StringSerialized<Cid>`?

Reply:
nah, I just forgot about it; I'll try changing it